File: common/src/main/java/org/sandboxpowered/silica/network/play/serverbound/TeleportConfirmation.java
package org.sandboxpowered.silica.network.play.serverbound;

import org.sandboxpowered.silica.network.PacketByteBuf;
import org.sandboxpowered.silica.network.PacketHandler;
import org.sandboxpowered.silica.network.PacketPlay;
import org.sandboxpowered.silica.network.PlayContext;

public class TeleportConfirmation implements PacketPlay {
    private int tpId;

    @Override
    public void read(PacketByteBuf buf) {
        tpId = buf.readVarInt();
    }

    @Override
    public void write(PacketByteBuf buf) {

    }

    @Override
    public void handle(PacketHandler packetHandler, PlayContext context) {

    }
}

File: common/src/main/kotlin/org/sandboxpowered/silica/network/play/serverbound/TeleportConfirmation.kt
package org.sandboxpowered.silica.network.play.serverbound

import org.sandboxpowered.silica.network.PacketByteBuf
import org.sandboxpowered.silica.network.PacketHandler
import org.sandboxpowered.silica.network.PacketPlay
import org.sandboxpowered.silica.network.PlayContext

class TeleportConfirmation(private var tpId: Int = 0) : PacketPlay {

    override fun read(buf: PacketByteBuf) {
        tpId = buf.readVarInt()
    }

    override fun write(buf: PacketByteBuf) {}
    override fun handle(packetHandler: PacketHandler, context: PlayContext) {}
}